### PR TITLE
[14.0][IMP] base_rest: Add new CerberusListValidator used to support array …

### PR DESCRIPTION
**Pull request by lmignon**
…of json validated by cerberus as request or response body
image

### CODE REVIEW

**Comment by lmignon** [comment]
ping @simahawk @sebastienbeau

**Comment by lmignon** [comment]
ping @acsonefho @rousseldenis @kevinkhao

File: base_rest/restapi.py [hunk 1]
```diff
@@ -188,18 +188,14 @@ def _to_http_response(self, result):
 
 
 class CerberusValidator(RestMethodParam):
-    def __init__(self, schema=None, is_list=False):
+    def __init__(self, schema, is_list=False):
```
Comment (simahawk) [inline base_rest/restapi.py]:
```suggestion
    def __init__(self, schema, **kw):
```

**Review by simahawk — COMMENTED** [review]
to be documented in README?

**Review by acsonefho — APPROVED** [review]

File: base_rest/restapi.py (hunk 1 above)
Comment (lmignon) [reply base_rest/restapi.py]: Why would you want to have '**kw' as an argument. IMO it's useless and not safe. Here we know what is needed for the class and these python classes are not extensible by other addons.

**Comment by lmignon** [comment]
> to be documented in README?

Would be nice but IMO readme is not the right format to provide the documentation for the API. It's not allowed to declare subsections into readme parts and if we document this validator, we should also provides the documentation for the whole API which is outside the scope of this RP.

File: base_rest/restapi.py (hunk 1 above)
Comment (simahawk) [reply base_rest/restapi.py]: Yeah, not the best indeed but I thought you were willing to support other args, my bad.

Anyway `is_list` is useless too, why shall you keep it?
For backward compat? If so, then you should log is deprecated, no? Or at least add a comment explaining why is there.
If not, just remove it.

**Comment by simahawk** [comment]
> > to be documented in README?
> 
> Would be nice but IMO readme is not the right format to provide the documentation for the API. It's not allowed to declare subsections into readme parts and if we document this validator, we should also provides the documentation for the whole API which is outside the scope of this RP.

AFAIK yes, you can have a sub section using `~~~`.
You already have an example in README regarding the normal validator, why should it be different?

**Comment by lmignon** [comment]
@simahawk done

File: base_rest/restapi.py (hunk 1 above)
Comment (lmignon) [reply base_rest/restapi.py]: removed... (My bad, I had forgotten to delete it)

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
remember to squash ;)

**Comment by lmignon** [comment]
> remember to squash ;)

it's done now that your review is completed ;-)

**Pull request merged**